src/Compilers/CSharp/Portable/CSharpResources.resx

```diff
@@ -5731,6 +5731,9 @@ To remove the warning, you can use /reference instead (set the Embed Interop Typ
   <data name="IDS_FeatureNullPointerConstantPattern" xml:space="preserve">
     <value>null pointer constant pattern</value>
   </data>
+  <data name="IDS_FeatureDefaultTypeParameterConstraint" xml:space="preserve">
+    <value>default type parameter constraints</value>
```

Comment:
>default type parameter constraints [](start = 11, length = 34)

It feels like the name of the feature isn't very clear, especially that `default` is used in not very common cases. For example, a user typed ```T?``` and compiler reports that "default type parameter constraints" aren't supported by the language version. What constraints the message is talking about? This isn't a constraint.

Reply:
Perhaps we need another feature name for ```T?```

---
In reply to: [456602711](https://github.com/dotnet/roslyn/pull/45993#discussion_r456602711) [](ancestors = 456602711)

Reply:
This message is only reported for uses of `where T : default`. For uses of `T?`, we continue to report `ErrorCode.ERR_NullableUnconstrainedTypeParameter`.

---
In reply to: [456603253](https://github.com/dotnet/roslyn/pull/45993#discussion_r456603253) [](ancestors = 456603253,456602711)